Added 'TagString' test

patilatharva opened the pull request:
This test ensures tag objects have the string representation: Tag(time=$time, microstep=$microstep)

Soroosh129 commented:
Hi @patilatharva 

It looks like the test you've added is filing with the following error:
```Python
expected tag string representation: Tag(time=$time, microstep=$microstep), instead got: {tag_str}
```
Do you know why that might be?

patilatharva commented:
This seems to be because the python target submodule wasn't updated. The git history is a bit messy now but I just updated so the test should pass. Also, thanks to your comment I noticed a formatting error in the error message, which I fixed as well.

Soroosh129 commented:
> This seems to be because the python target submodule wasn't updated. 

Ah I see.

> The git history is a bit messy now.

No worries. We an squash the commits. Thanks for adding this feature!

lhstrh commented:
@patilatharva: I looks like this PR has been under the radar for a while. Shall we bring it back into a state to where we can merge it? One issue with this PR is that I don't quite understand what it attempts to do, so if you could update the description to clarify, that would be a great first step. Thanks, and sorry for the delayed handling of this!

patilatharva commented:
Yes, thanks for bringing this back to my attention! I'll make it ready for review.
